code/datums/elements/deployable_item.dm

```diff
@@ -28,13 +29,14 @@
 /datum/element/deployable_item/Detach(datum/source, force)
 	. = ..()
 	UnregisterSignal(source, COMSIG_ITEM_EQUIPPED)
+	LAZYREMOVE(GLOB.deployable_items[source.type], source)
 
 ///Register click signals to be ready for deploying
 /datum/element/deployable_item/proc/register_for_deploy_signal(obj/item/item_equipped, mob/user, slot)
 	SIGNAL_HANDLER
 	if(slot != SLOT_L_HAND && slot != SLOT_R_HAND)
 		return
-	RegisterSignal(user, COMSIG_MOB_MOUSEDOWN, .proc/deploy)
+	RegisterSignal(user, COMSIG_MOB_MOUSEDOWN, .proc/deploy, TRUE)
```

Comment:
You need to justify this

Reply:
Holding two deployables

Reply:
No with a comment

Reply:
Also should probably be a component but w/e

Reply:
Spooky said keep it as an element. Added the comment though.